src/Crypto/ECKeyPair.php

```diff
@@ -155,7 +155,7 @@ public function signature(string $hexData): ?string
             throw new \Exception('Could not construct a recoverable keyPair. This should never happen.');
         }
 
-        $canonicalSignature = dechex($finalRecId + 31) . $signatureHex;
+        $canonicalSignature = gmp_strval(gmp_init($finalRecId + 31, 10), 16) . $signatureHex;
```

Comment:
Would suggest using one coherent method in `Utils/Math.php`, something like:
```
public static function gmpIntToStr($number): string {
    return gmp_strval(gmp_init($number, 10), 16);
}
```

Example use:
```suggestion
        $canonicalSignature = Math::gmpIntToStr($finalRecId + 31) . $signatureHex;
```